litho-widget/src/main/java/com/facebook/litho/widget/RecyclerBinder.java

```diff
@@ -2704,14 +2716,22 @@ void initRange(
                 logger.newPerformanceEvent(mComponentContext, EVENT_INIT_RANGE));
 
     try {
-      final Size size = new Size();
-      holder.computeLayoutSync(mComponentContext, childWidthSpec, childHeightSpec, size);
+      if (mInitRangeSize > 0 && ThreadUtils.isMainThread()) {
```

Comment:
Just a guess, since UNSET == -1

```suggestion
      if (mInitRangeSize != UNSET && ThreadUtils.isMainThread()) {
```

Reply:
The initial code is correct, we also need mInitRangeSize to not be zero. If it is zero, there will be a division by zero in lines 2722 and 2723